### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by AkintoyeOlamide**
This project is was simply on how forms are created, edited, and submitted with rails.
And this project was made rails version 6.0.3 and ruby version 2.7.1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,52 @@
...
+
+**To get this project set up on your local machine, follow these simple steps:**
+
+1. Open Terminal.
+2. Navigate to your desired location to download the contents of this repository.
+3. Copy and paste the following code into the Terminal: git clone https://github.com/LamiaSristy/re-former
+4. Run ```cd re-former```.
+5. Run ```bundle install``` to get the necesary gems.
+6. Run `rails db:create`.
+6. Run `rails db:migrate`.
```
Comment (javitocor) [inline README.md]: hen using sqlite you do not need to run db:create, kindly remove it

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<%# <form action="/users" method="post" accept-charset="UTF-8">
+  <label for="name">Username:</label>
```
Comment (javitocor) [inline app/views/users/new.html.erb]: Modify your action to meet the rails path, ie action="users_path"

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  <input type="password" id="password" name="user[password]"><br><br>
+  <input type = "hidden" name = "authenticity_token" value = "<%=form_authenticity_token%>
+  <%# <input type="submit" value="Submit"> %>
+<%# </form> %>
+
+
+<%
+=begin %>
+<%= form_tag('/users') do %> 
+  <%= label_tag 'username', 'Username:'%>
```
Comment (javitocor) [inline app/views/users/new.html.erb]: Same here

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+<%# </form> %>
+
+
+<%
+=begin %>
+<%= form_tag('/users') do %> 
+  <%= label_tag 'username', 'Username:'%> 
+  <%= text_field_tag 'username'%> 
+  <%= label_tag 'email', 'Email:'%> 
+  <%= text_field_tag 'email'%>
```
Comment (javitocor) [inline app/views/users/new.html.erb]: Use appropriate tag for the email field

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  <%= submit_tag 'submit' %>
+<% end %>
+<%
+=end %>
+
+
+<%= form_for @user, url: {action: "create"}, html: {class: ""} do |f| %>
+  UserName: <%= f.text_field :username %> 
+  Email: <%= f.text_field :email, placeholder: 'example@example.com' %>
+  Password: <%= f.text_field :password, placeholder: '**********' %>
```
Comment (javitocor) [inline app/views/users/new.html.erb]: same here, use appropriate field for the email

File: app/views/users/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+  <% if @user.errors.any? %>
+  <% @user.errors.full_messages.each do |error| %>
+    <%= error %>
+  <% end %>
+  <% end %>  
+
+  UserName: <%= f.text_field :username %>
+  Email: <%= f.text_field :email, placeholder: 'example@example.com' %>
+  Password: <%= f.text_field :password, placeholder: '**********' %>
```
Comment (javitocor) [inline app/views/users/edit.html.erb]: Same here

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED ♻️ 

A great job so far with this project! 👍 
Just a few adjustments and ready to go 💪 
Kindly check my comments below 👇  apply the changes and resubmit your project for a new code review ⏩ 
Good luck 😃 

## Mandatory changes
- Commit more often and be more descriptive in your commits, it will help you out to know your trip from the beginning till the end of the project
- Remove all commented out code form your controller

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED 🛑 

A great job with the implemented changes 👍 
But not all of them yet, be sure to implement all the mandatory changes before resubmitting your project
Good luck 😃 

## Mandatory changes
- Kindly make the development branch your default branch
- Your forms form_tag and form_for still does not have the appropriate field for the email, kindly implement it
- Kindly add the 'users_path' to your form_tag form
- In the readme file, kindly modify the db:create by db:migrate in your getting started section
- Modify the email field also in your users/edit.erb

**Comment by AkintoyeOlamide** [comment]
The new changes that were required by Javier were, were things the previous tse instructed me to do, so it's like the reviews are conflicting with each other.

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED 🛑 

Hello Akintoye!
Good job so far! But are some mandatory changes you have to implement that are not yet done!
Let's review them again 💪  apply the changes and request for a new code review ⏩ 

## Mandatory changes
- On your form_tag and in the form_for, in the users new and edit views, apply the following changes. Those changes are necessary in order to apply the appropriate input field for the user, besides all of them comes with extra security features, ie. by default will check if the email provided is valid
    - uses label_tag, text_field_tag, email_field_tag, password_field_tag, for the form_tag
    - `<%= f.email_field :email %>` for the form_for

**Review by rOluochKe — APPROVED** [review]
Hello @AkintoyeOlamide 
### You got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**